packages/gamut/src/Popover/__tests__/Popover-test.tsx

```diff
@@ -0,0 +1,173 @@
+  });
+
+  it('shows a screen overlay if the prop is true', () => {
+    renderPopover({
+      isOpen: true,
+      showScreen: true,
+    });
+
+    expect(screen.queryByTestId('popover-screen')).toBeInTheDocument();
+  });
+
+  it('does not show a beak if the prop is false', () => {
+    renderPopover({
+      isOpen: true,
+      showBeak: false,
+    });
+
+    expect(screen.queryByTestId('popover-beak')).not.toBeInTheDocument();
+  });
+
+  it('shows a beak if the prop is true', () => {
+    renderPopover({
+      isOpen: true,
+      showBeak: true,
+    });
+
+    expect(screen.queryByTestId('popover-beak')).toBeInTheDocument();
+  });
+
+  it("positions with default 'below', 'left', '20' value when position, align, offset props are not provided respectively", () => {
+    Object.defineProperty(window, 'scrollY', { value: 1 });
+    Object.defineProperty(window, 'scrollX', { value: 1 });
+
+    const wrapped = mountPopover({
+      isOpen: true,
+    });
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('top', 319);
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('left', 58);
+  });
+
+  it('positions with given position and align values when provided', () => {
+    Object.defineProperty(window, 'scrollY', { value: 1 });
+    Object.defineProperty(window, 'scrollX', { value: 1 });
+
+    const wrapped = mountPopover({
+      isOpen: true,
+      position: 'above',
+      align: 'right',
+    });
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('top', 241);
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('left', 841);
+  });
+
+  it('positions with given offset value when provided', () => {
+    Object.defineProperty(window, 'scrollY', { value: 1 });
+    Object.defineProperty(window, 'scrollX', { value: 1 });
+
+    const wrapped = mountPopover({
+      isOpen: true,
+      position: 'above',
+      align: 'right',
+      offset: 30,
+    });
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('top', 231);
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('left', 841);
+  });
+
+  it('positions round to whole number when ', () => {
+    Object.defineProperty(window, 'scrollY', { value: 1.5 });
+    Object.defineProperty(window, 'scrollX', { value: 1.5 });
+
+    const wrapped = mountPopover({
+      isOpen: true,
+      position: 'above',
+      align: 'right',
+      offset: 30,
+    });
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('top', 232);
+
+    expect(
+      wrapped.find('[data-testid="popover-content-container"]').prop('style')
+    ).toHaveProperty('left', 842);
```

Comment:
Nit: by combining into one assertion, we get a more informative error message that can tell us whether _both_ or _just one_ of these are failing:

```suggestion
    expect(wrapped.find('[data-testid="popover-content-container"]').props()).toMatchObject({
      style: {
        top: 231,
        left: 841,
      }
    });
```

Up for using this here and elsewhere?

I'm also a little surprised at these random numbers being stable. I guess the test environment just always happens to have the same rendering logic? 🤷 if it works it works.

Reply:
Yes! i can look at combining to assertions.